portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,32 +16,52 @@
 
 
 import com.google.sps.data.Comment;
+import com.google.sps.data.CommentsList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import java.util.Arrays;
-import com.google.gson.Gson;
-import java.lang.reflect.Type;
-import com.google.gson.reflect.TypeToken;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  private ArrayList<Comment> comments = new ArrayList<Comment>(Arrays.asList(
-      new Comment("user1", "Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur"),
-      new Comment("user2", "Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur"),
-      new Comment("user3","Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur Lorem ipsum dolor sit amet, consectetur")
-    ));
+
+  private CommentsList comments = new CommentsList(); 
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {    
-    String json = convertCommentsToJson(this.comments);
+    //reading GET request max-comments parameter to determine how much comments include in the response
+    String commentsAction = getParameter(request, "comments", "current");
+    String newestStr = getParameter(request, "newest", "");
+    String oldestStr = getParameter(request, "oldest", "");
+    String jsonArray = "";
+    System.out.println(newestStr);
+    System.out.println(oldestStr);
+    long oldest = Long.MAX_VALUE;
+    long newest = 0;
+    try{
+      oldest = Long.parseLong(oldestStr);
+      newest = Long.parseLong(newestStr);
+    }catch(NumberFormatException e){
+      System.out.println("Cant parse");
```

Comment:
This error message doesn't help debugging. Instead simply print the exception to get more info.